pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1117,14 +1115,7 @@ impl Timeline {
         // Under normal circumstances, we will accumulate up to compaction_interval L0s of size
         // checkpoint_distance each.  To avoid edge cases using extra system resources, bound our
         // work in this function to only operate on this much delta data at once.
```

Comment:
nit: comment needs updating, we're now bounding it at `compaction_upper_limit` rather than `compaction_interval`.

Reply:
added in https://github.com/neondatabase/neon/pull/10564